### ISSUE TITLE
TestAdapter not discovered by TFS2015 update 3

### ISSUE THREAD

**Issue by Liero** [mmanela/chutzpah#556]
I'm trying to use chutzpah just to enable the test on TFS 2015 (update 2) build server, but I ended up with:

> Warning: No test is available in the provided sources. Make sure that installed test discoverers & executors, platform & framework version settings are appropriate and try again

I have app written using VS Code and nodejs. There is no .sln file.  I've added nuget.config and packages.config to my root folder and installed the nuget package. On the buils server there is nuget restore build step and the packages are restored, however the test adapter is not recognized.

When I try it locally on my dev machine, the test adapter are not recognized either:

```
>"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" /TestAdapterPath:"C:\Temp\unit testing\Packages\"
Microsoft (R) Test Execution Command Line Tool Version 14.0.25123.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Warning: The path 'C:\Temp\unit testing\Packages\' specified in the 'TestAdapterPath' does not contain any test adapters, provide a valid path and try again.
```

I believe that the assembly that contains test adapter must be named **_AdapterName_.TestAdapter.dll** in order to be discovered.

_\* I can provide source code if needed_

[unittesting.zip](https://github.com/mmanela/chutzpah/files/467374/unittesting.zip)

**Comment by mmanela** [comment]
So I tested locally and it works fine. I created a new folder. Did ```nuget install chutzpah``` and then ran:

```
& "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" /TestAdapterPath:"c:\dev\adtest" .\test\test.js
```

And it worked fine. Are you doing something different?

**Comment by Pablissimo** [comment]
I was seeing this with 4.2.4, fixable by explicitly setting the path to Chutzpah as you have. c.f. with the NUnit or XUnit adapter NuGet packages for example , which are automatically discovered by the build (i.e. I don't need to set the 'Path to custom test adapters' setting for an XUnit/NUnit test step, but do for Chutzpah).

However, on 4.3.3 and with a TFS upgrade (unclear which fixed it) it seems now to be working without explicitly setting the path to the adapter as part of the build - @Liero are you still having trouble?

**Comment by Liero** [comment]
After I upgraded Chutzpah to 4.3.3 I was able to discover test adapters. When running localy I had to specify  `/TestAdapterPath:"C:\temp\unittesting\"`, I haven't tried on TFS. 

Thanks

**Comment by Slion** [comment]
When implementing my own Test Adapter for VsTest I ended up in a a situation where it would work locally when the DLL was copied to the Visual Studio extensions folder but did not work on VSTS or when using the `/TestAdapterPath` option locally.

It turns out that for VSTS and `/TestAdapterPath` to work your adapter binary name must end with `.TestAdapter.dll`.

Nothing to do with chutzpah really but I landed here after searching online for solution to that problem.
